agent/taskresource/fsxwindowsfileserver/fsxwindowsfileserver_windows.go

```diff
@@ -519,9 +515,34 @@ func (fv *FSxWindowsFileServerResource) retrieveFileSystemDNSName(fileSystemId s
 	return nil
 }
 
+var mountLock sync.Mutex
+var DriveLetterAvailable = utils.IsAvailableDriveLetter
 var execCommand = exec.Command
 
-func (fv *FSxWindowsFileServerResource) performHostMount(remotePath string, localPathArg string, username string, password string) error {
+func (fv *FSxWindowsFileServerResource) performHostMount(remotePath string, username string, password string) error {
+	mountLock.Lock()
```

Comment:
can you add a comment here documenting why a package level lock is needed here? so that we don't lose context on why this was added

Reply:
does this delay the resource creation for fsx tasks?

Reply:
I ran 10 tasks together and noticed about half a second delay between tasks transitioning into running state.

Reply:
I'll add a comment to add some context